sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusAsyncConsumer.java

```diff
@@ -34,9 +34,7 @@
         this.linkProcessor = linkProcessor;
         this.messageSerializer = messageSerializer;
         this.processor = linkProcessor
-            .map(message -> this.messageSerializer.deserialize(message, ServiceBusReceivedMessage.class))
-            .publish(receiverOptions.getPrefetchCount())
-            .autoConnect(1);
```

Comment:
Why was this removed?

Reply:
This is removed because the publish( ) method requests the upstream (link processor) with its own request count (the reactor prefetch). Without publish, if a user's code subscriber request n records, the link processor can adjust the link credits with by the user's request count. So with this publish(), we won't be able to adjust the link credits by the user's request count.
Plus, receiverOptions.getPrefetchCount() can be 0 now and defaults to be 0. It's an invalid value for publish() method.

The side effect of no-publish is a receiveMessages can't be published twice. The link processor will throw an error if there are two subscribers. 
We need to discuss offline.